File: cp_rust/src/bin/cses-1621-distinct-numbers.rs
// Created by Ayush Biswas at 2025/07/14 15:11
// https://cses.fi/problemset/task/1621
use cp_lib::*;

// @code begin
use cpio::*;
use itertools::Itertools;

sol! {
    fn solution(
        _n: usize,
        a: [usize]
    ) -> usize {
        a.into_iter().unique().count()
    }
}

// @code end

File: cp_haskell/bin/cses-1621-distinct-numbers.hs
-- Created by Ayush Biswas at 2026/03/11 19:26
-- https://cses.fi/problemset/task/1621

-- @code begin

import Data.ByteString.Char8 qualified as BS
import Data.Maybe (mapMaybe)
import Data.Set qualified as Set

getInts :: IO [Int]
getInts = mapMaybe (fmap fst . BS.readInt) . BS.words <$> BS.getLine

main :: IO ()
main = do
  n :: Int <- read <$> getLine
  xs :: [Int] <- getInts
  let res = length $ Set.fromList xs
  print res

-- @code end